### PR TITLE
Upgrade PHPStan and PHP-CS-Fixer verisons

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log

- Upgrading PHPStan to be `^1` version.
- Upgrading PHP-CS-Fixer to be `^3` version.

### CODE REVIEW

**Comment by odan** [comment]
Thanks, but please do not remove the return type in the DocBlocks.

File: .cs.php [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
     //->setCacheFile(__DIR__ . '/.php_cs.cache')
     ->setRules([
         '@PSR1' => true,
         '@PSR2' => true,
         '@Symfony' => true,
-        'psr4' => true,
+        'psr_autoloading' => true,
         // custom rules
         'align_multiline_comment' => true, // psr-5
+        'no_superfluous_phpdoc_tags' => false,
```
Comment (peter279k) [inline .cs.php:14]: > Thanks, but please do not remove the return type in the DocBlocks.

Adding this setting to let the `return type hint annotation` be passed without checking.

File: tests/CsvWriterTest.php [hunk 2]
```diff
@@ -19,7 +19,7 @@ final class CsvWriterTest extends TestCase
     /**
      * @var vfsStreamDirectory
      */
-    private $root;
```
Comment (odan) [inline tests/CsvWriterTest.php:22]: This must be a private property, because it is a final class.

Comment (peter279k) [reply tests/CsvWriterTest.php:22]: This is reported by PHPStan:

```
> phpstan analyse src tests --level=max -c phpstan.neon --no-progress --ansi
 ------ ----------------------------------------------------------------------- 
  Line   tests/CsvWriterTest.php                                                
 ------ ----------------------------------------------------------------------- 
  22     Property Selective\Test\CsvWriterTest::$root is never read, only       
         written.                                                               
         💡 See:                                                                 
         https://phpstan.org/developing-extensions/always-read-written-propert  
         ies                                                                    
 ------ ----------------------------------------------------------------------- 


 [ERROR] Found 1 error                                                          
```

Should we ignore this issue?

Comment (odan) [reply tests/CsvWriterTest.php:22]: No, but this should be fixed in another PR.

Comment (peter279k) [reply tests/CsvWriterTest.php]: Reverted and it should expected to be failed on GitHub action running.

**Pull request merged**